### ISSUE TITLE
add support swift 6.0 with  "-strict-concurrency=complete"

### ISSUE THREAD

**Issue by rainliu** [apple/swift-nio#2751]
### Expected behavior
SwiftNIO should support swift 6.0 build with "-strict-concurrency=complete"

### Actual behavior
It failed to build: 
https://swiftpackageindex.com/apple/swift-nio/builds

**Comment by Lukasa** [comment]
Yes, this is on our roadmap. However, I don’t think it blocks any customer adoption, so for now we’re not rushing to get it done by the Swift 6 ship date.
